hydroflow/src/progress/mod.rs

```diff
@@ -0,0 +1,88 @@
+// TODO(justin): For now, require these to be totally ordered. We should make
+// them lattices later, though.
+pub trait Timestamp: Clone + Ord + std::fmt::Debug {
+    fn min() -> Self;
+    fn max() -> Self;
+    fn le(&self, other: &Self) -> bool;
```

Comment:
Probably makes sense to eventually change this to `partial_cmp(...) -> Option<std::cmp::Ordering>` or equivalent